core/includes/database/mysql/database.inc

```diff
@@ -127,6 +128,25 @@ class DatabaseConnection_mysql extends DatabaseConnection {
     return 'mysql';
   }
 
+  /**
+   * Overrides DatabaseConnection::createDatabase().
+   *
+   * @param string $database
+   *   The name of the database to create.
+   *
+   * @throws DatabaseNotFoundException
+   */
+  public function createDatabase($database) {
+    try {
+      // Create the database and set it as active.
+      $this->exec("CREATE DATABASE $database");
```

Comment:
@quicksketch should we put a `IF NOT EXISTS` on this?
https://dev.mysql.com/doc/refman/5.7/en/create-database.html

Reply:
I don't think adding it would do any harm. Though I don't think we'd have any situation where that should occur.

Reply:
I've updated the PR to include that.